packages/core/types/index.d.ts

```diff
@@ -24,6 +24,8 @@ export {
   ObjectInterpolation
 } from '@emotion/css'
 
+export type AnyIfEmpty<T extends object> = keyof T extends never ? any : T
```

Comment:
I think this was the sort of type which caused the number of types and build time to increase dramatically during compilation.

It would be worth trying to see how many types are created before/after during a compilation with --diagnostics